### PR TITLE
Improved Docstring of scipy.special.hyp2f1

### PR DESCRIPTION
**Pull request by ashwinpathak20**
I have made a new pull request for the issue #6966 , please have a look at it and suggest me the changes.

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
     """)
 
 add_newdoc("scipy.special", "hyp2f1",
-    """
+    r"""
     hyp2f1(a, b, c, z)
 
-    Gauss hypergeometric function 2F1(a, b; c; z).
+    Gauss hypergeometric function 2F1(a, b; c; z), defined as:
+
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: This needs a ``.. math::`` directive above it surrounded by blank lines. And then indent the LaTeX.

Comment (rgommers) [inline scipy/special/add_newdocs.py]: Also, keep all lines to <80 characters.

Comment (rgommers) [inline scipy/special/add_newdocs.py]: and same for the other LaTeX expressions

File: scipy/special/add_newdocs.py [hunk 2]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+    \mathrm{hyp2f1}(a, b, c, z) = 1 + \sum_{k=0}^inf. \frac {{a+k}\choose{a}} {b+k}\choose{b}}{{c+k}\choose{c}} z^(k+1).
+
+    OR  
+
+    \mathrm{hyp2f1}(a, b, c, z) = \sum_{n=0}^inf. \frac{(a)_n (b)_n}{(c)_n}\frac{z^n}{n!}
+
+     where :math:`(\cdot)_n` is the Pochhammer symbol; see `poch`. When
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: indentation is off here, this line is 5 spaces

File: scipy/special/add_newdocs.py [hunk 3]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+    :math:`n` is an integer the result is a polynomial of degree
+    :math:`n`
+
+    Gaussian hypergeometric function is defined on the disk |z|<1, and by analytic continuation elsewhere.
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: double backticks around ``|z| < 1``, this is what caused the CI failure.

File: scipy/special/add_newdocs.py [hunk 4]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+    The branch obtained by introsucing a cut from 1 to infinity on the real z-axis, 
+    that is, the branch in the sector |ph(1-z)| <= \pi, is the principal branch (or principal value) of F(a,b;c;z).
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: same here, double backticks. Note that ``|  |`` is a special directive in Sphinx, so you always need to typeset this as code.

File: scipy/special/add_newdocs.py [hunk 5]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+
+    Parameters
+    ----------
+    a : real array_like
+        argument
+
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: no blank lines between parameters

File: scipy/special/add_newdocs.py [hunk 6]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+    :math:`n` is an integer the result is a polynomial of degree
+    :math:`n`
+
+    Gaussian hypergeometric function is defined on the disk |z|<1, and by analytic continuation elsewhere.
+    The branch obtained by introsucing a cut from 1 to infinity on the real z-axis, 
+    that is, the branch in the sector |ph(1-z)| <= \pi, is the principal branch (or principal value) of F(a,b;c;z).
+
+    Parameters
+    ----------
+    a : real array_like
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: just ``array_like``

Comment (rgommers) [inline scipy/special/add_newdocs.py]: same for the other parameters. this is a type specifier, we don't add value indicators like real normally. If it's important, add it to the description instead.

File: scipy/special/add_newdocs.py [hunk 7]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+
+    c : real array_like
+        argument
+
+    z : real or complex array_like
+        argument
+
+    Returns
+    -------
+        ndarray ,
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: Return value needs a name. You can use:

    hyp2f1 : scalar or ndarray

File: scipy/special/add_newdocs.py [hunk 8]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+    c : real array_like
+        argument
+
+    z : real or complex array_like
+        argument
+
+    Returns
+    -------
+        ndarray , 
+        the values of the gaussian hypergeometric function.
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: nitpick: start line with a capital letter

File: scipy/special/add_newdocs.py [hunk 9]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+
+    Notes
+    -----
+
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: no blank line here

File: scipy/special/add_newdocs.py [hunk 10]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+    hyp2f1 can be used to find the values of the gaussain hypergeometric function.
+
+    http://dlmf.nist.gov/15.2
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: why is this link here?

Comment (rgommers) [inline scipy/special/add_newdocs.py]: same question for other links below

File: scipy/special/add_newdocs.py [hunk 11]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+        the values of the gaussian hypergeometric function.
+
+    Notes
+    -----
+
+    hyp2f1 can be used to find the values of the gaussain hypergeometric function.
+
+    http://dlmf.nist.gov/15.2 
+
+    For complex arguments z with |z| >= 1 it can be analytically continued along any path in the complex plane that avoids the branch points 0 and 1.
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: double backticks again

File: scipy/special/add_newdocs.py [hunk 12]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+
+    As c tends to -m, where m is a positive integer, 2F1(a,b;c;z) tends to infinity.
+
+    http://dlmf.nist.gov/15.2 
+
+    Cases addressed are
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: this needs to be a non-indented list (so ``- `` symbols)

File: scipy/special/add_newdocs.py [hunk 13]
```diff
@@ -3036,10 +3036,86 @@ def add_newdoc(place, name, doc):
...
+      Tests and escapes for negative integer a, b, or c
+      Linear transformation if c - a or c - b negative integer
+      Special case c = a or c = b
+      Linear transformation for  z near +1
+      Transformation for z < -0.5
+      Psi function expansion if z > 0.5 and c - a - b integer
+      Conditionally, a recurrence on c to make c-a-b > 0
+
+      z < -1  AMS 15.3.7 transformation applied (Travis Oliphant)
```
Comment (rgommers) [inline scipy/special/add_newdocs.py]: ``z < -1`` needs to be typeset as code. don't mention names of people in docs.

File: scipy/special/add_newdocs.py (hunk 10 above)
Comment (ashwinpathak20) [inline scipy/special/add_newdocs.py]: Shall I include these links in the reference then?

File: scipy/special/add_newdocs.py (hunk 13 above)
Comment (ashwinpathak20) [inline scipy/special/add_newdocs.py]: What exactly do you mean by typeset as code?

File: scipy/special/add_newdocs.py (hunk 2 above)
Comment (ashwinpathak20) [inline scipy/special/add_newdocs.py]: what do you mean by this line is 5 spaces?

File: scipy/special/add_newdocs.py (hunk 12 above)
Comment (pv) [inline scipy/special/add_newdocs.py]: I don't think it's useful to explain the algorithm here in this level of detail, it's too complex, so it should just give references. If there are restrictions on parameter ranges, those can be explained.

At most, "Real-valued case is based on modified Cephes [1]_ routine `hyp2f1`, complex-valued case ...".

Comment (ashwinpathak20) [inline scipy/special/add_newdocs.py]: Ok, so I need to remove the cases part from Notes section and instead just give references ?

**Comment by rgommers** [comment]
Updated this PR, good to go now I think.

File: scipy/special/add_newdocs.py [hunk 14]
```diff
@@ -3365,10 +3365,52 @@ def add_newdoc(place, name, doc):
...
+
+    .. math::
+
+       \mathrm{hyp2f1}(a, b, c, z) = \sum_{n=0}^\infty
+       \frac{(a)_n (b)_n}{(c)_n}\frac{z^n}{n!},
+
+    where :math:`(\cdot)_n` is the Pochhammer symbol; see `poch`. When
+    :math:`n` is an integer the result is a polynomial of degree :math:`n`.
+
+    `hyp2f1` is defined on ``|z| < 1`` and by analytic continuation elsewhere.
```
Comment (person142) [inline scipy/special/add_newdocs.py]: Might read better if it says "This function is defined for ``|z| < 1`` as [equation], and defined on the rest of the complex ``z``-plane by analytic continuation. Here ``(-)_n`` is the..."

Comment (rgommers) [inline scipy/special/add_newdocs.py]: good point, done.

**Pull request merged**